src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -179,7 +184,17 @@ public void Dispose()
                         {
                             concurrencyLimiter.Release();
                         }
-                    }, cancellationToken, TaskCreationOptions.AttachedToParent,  TaskScheduler.Default);
+                    }, cancellationToken);
+
+                    task.ContinueWith(t =>
```

Comment:
Perhaps a comment to explain why this can't be combined with the Task.Run ? (since we'll be screwed if someone merges them back?)